cmd/gardener-extension-provider-local/main.go

```diff
@@ -27,11 +26,10 @@ import (
 
 func main() {
 	utils.DeduplicateWarnings()
-
-	logf.SetLogger(logger.MustNewZapLogger(logger.InfoLevel, logger.FormatJSON))
```

Comment:
We should keep this for now. For the provider local the logger isn't set anywhere else so https://github.com/gardener/gardener/blob/e7e85b27f472ab2f4257f49c2fd7f5495218958a/cmd/gardener-extension-provider-local/app/app.go#L207 a bit later on, will actually end up with a `NullLogSink`

When running it locally I got (notice the last line):
```{"level":"info","ts":"2024-03-26T06:49:53.405Z","msg":"Getting rest config for garden"}
{"level":"info","ts":"2024-03-26T06:49:53.406Z","msg":"Setting up cluster object for garden"}
{"level":"info","ts":"2024-03-26T06:49:53.406Z","msg":"Adding garden cluster to manager"}
{"level":"info","ts":"2024-03-26T06:49:53.406Z","msg":"Adding controllers to manager"}
{"level":"info","ts":"2024-03-26T06:49:53.470Z","msg":"Started with","hostIP":"127.0.0.1"}
{"level":"info","ts":"2024-03-26T06:49:53.473Z","msg":"starting server","kind":"health probe","addr":"[::]:8081"}
I0326 06:49:53.474271       1 leaderelection.go:250] attempting to acquire leader lease extension-provider-local-jxjbc/provider-local-leader-election...
{"level":"info","ts":"2024-03-26T06:49:53.474Z","msg":"Starting EventSource","controller":"webhook-certificate-reloader","source":"func source: 0x16a76a0"}
{"level":"info","ts":"2024-03-26T06:49:53.474Z","msg":"Starting Controller","controller":"webhook-certificate-reloader"}
{"level":"info","ts":"2024-03-26T06:49:53.474Z","msg":"Starting workers","controller":"webhook-certificate-reloader","worker count":1}
{"level":"info","ts":"2024-03-26T06:49:53.575Z","msg":"Found new secret, writing certificate to disk","controller":"webhook-certificate-reloader","object":{"name":""},"namespace":"","name":"","reconcileID":"afc88ad0-811a-493b-ab75-7a58076e42b1","secretConfigName":"provider-local-webhook-server","secretNamespace":"extension-provider-local-jxjbc","identity":"gardener-extension-provider-local-webhook","certDir":"/tmp/gardener-extensions-cert","secretName":"provider-local-webhook-server-b3a809f3"}
[controller-runtime] log.SetLogger(...) was never called; logs will not be displayed.
```

Not sure if we could also use the [`InitRun`](https://github.com/gardener/gardener/blob/e7e85b27f472ab2f4257f49c2fd7f5495218958a/cmd/utils/run.go#L43) for the provider local similar to how it is done here:  https://github.com/gardener/gardener/blob/e7e85b27f472ab2f4257f49c2fd7f5495218958a/cmd/gardenlet/app/app.go#L89 to setup logging but this could be done in a separate PR.

Reply:
Seems like `utils.InitRun` is not being used consistently across all commands, as some commands use it, while others do not. I'll move the PR in a draft state, as this increases the scope of changes and will need more time to fully test everything.